### ISSUE TITLE
Fastcorrect：Enable to judge whether the output of  asr should be corrected or not?

### ISSUE THREAD

**Issue by YingjiaoLiu** [microsoft/NeuralSpeech#75]
I am thinking on how to let the Fastcorrect classify wrongly recognized tokens from all recognition results. That is, correct recogintion wouldn't feed into duration prediction and decoder. I added a binary classification layer after encoder to filter out tokens does not need to be corrected by setting a threshold, but I 'm not sure how to calcute the loss for this multi-task training. Could you help explain this? 
I noteiced you also used an auxiliary binary classification loss on each source token to judge whether this token is correct or not in https://github.com/microsoft/NeuralSpeech/issues/48#issuecomment-1191244703.

**Comment by YingjiaoLiu** [comment]
> About the auxiliary binary classification loss, you can reference to "[to_be_edited_loss](https://github.com/microsoft/NeuralSpeech/blob/master/FastCorrect/FastCorrect/fastcorrect_model.py#L312)".

Was this to_be_edited_loss calculated and used in your open-sourced pretrain model?  In my understanding, this loss was calculated based on each token.

 In my experiment, Fastcorrect should be able to judge whether **each input utterance** was recognized correctly or not. Wighted average loss with _the nll_loss_ https://github.com/microsoft/NeuralSpeech/blob/master/FastCorrect/FastCorrect/fc_loss.py#L63 would be a good choice? 

If so, I would add an additional layer to let Fastcorrect do this  binary classification. Do you recommend me to rewrite the code of encoder of just add a _Classifier Class_ outside the encoder like _DurationPredictor_?https://github.com/microsoft/NeuralSpeech/blob/802161c29c8f88b38fd5dee839fb7082fbd97179/FastCorrect/FastCorrect/fastcorrect_model.py#L456

**Comment by YingjiaoLiu** [comment]
I've designed a new model structure to do the pre-judgement.
